agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -429,11 +429,157 @@ impl DeriveSchema for Stage {
             Ok(result_doc)
         }
 
+        #[derive(Debug)]
+        struct ProjectPathNode {
+            pub children_nodes: LinkedHashMap<String, ProjectPathNode>,
```

Comment:
I know there's some gravity to using LinkedHashMap, but wondering if we actually need to here. I don't see any use of the linked list funcionality. Additionally and super super nitty, but it might be better to use `Option` to wrap to save a few bytes.